common/persistence/cassandraMetadataPersistence.go

```diff
@@ -176,89 +191,108 @@ func (m *cassandraMetadataPersistence) GetDomain(request *GetDomainRequest) (*Ge
 	var err error
 	info := &DomainInfo{}
 	config := &DomainConfig{}
-	if len(request.ID) > 0 {
-		if len(request.Name) > 0 {
-			return nil, &workflow.BadRequestError{
-				Message: "GetDomain operation failed.  Both ID and Name specified in request.",
-			}
-		}
+	replicationConfig := &DomainReplicationConfig{}
+	var replicationClusters []map[string]interface{}
+	var version int64
 
-		query = m.session.Query(templateGetDomainQuery,
-			request.ID)
-		err = query.Scan(
-			&info.ID,
-			&info.Name,
-			&info.Status,
-			&info.Description,
-			&info.OwnerEmail,
-			&config.Retention,
-			&config.EmitMetric)
-	} else if len(request.Name) > 0 {
-		query = m.session.Query(templateGetDomainByNameQuery,
-			request.Name)
-		err = query.Scan(
-			&info.ID,
-			&info.Name,
-			&info.Status,
-			&info.Description,
-			&info.OwnerEmail,
-			&config.Retention,
-			&config.EmitMetric)
-	} else {
+	if len(request.ID) > 0 && len(request.Name) > 0 {
+		return nil, &workflow.BadRequestError{
+			Message: "GetDomain operation failed.  Both ID and Name specified in request.",
+		}
+	} else if len(request.ID) == 0 && len(request.Name) == 0 {
 		return nil, &workflow.BadRequestError{
 			Message: "GetDomain operation failed.  Both ID and Name are empty.",
 		}
 	}
 
-	if err != nil {
+	handleError := func(name, ID string, err error) error {
+		identity := name
+		if len(ID) > 0 {
+			identity = ID
+		}
 		if err == gocql.ErrNotFound {
-			var d string
-			if len(request.ID) > 0 {
-				d = request.ID
-			} else {
-				d = request.Name
-			}
-
-			return nil, &workflow.EntityNotExistsError{
-				Message: fmt.Sprintf("Domain %s does not exist.", d),
+			return &workflow.EntityNotExistsError{
+				Message: fmt.Sprintf("Domain %s does not exist.", identity),
 			}
 		}
-
-		return nil, &workflow.InternalServiceError{
+		return &workflow.InternalServiceError{
 			Message: fmt.Sprintf("GetDomain operation failed. Error %v", err),
 		}
 	}
 
+	domainName := request.Name
+	if len(request.ID) > 0 {
+		query = m.session.Query(templateGetDomainQuery, request.ID)
```

Comment:
I think this change will lead to an additional call to get domain info. Why we cannot get all info with domainID?

Reply:
Cassandra do not support conditional update to multiple table.
Previously, we were using batch query to atomically update 2 tables, which (in my opinion) is an overkill, and cause a lot of unnecessary load on Cassandra.
This change will use the domains table just for domain uuid -> domain name lookup.
The actual content will be in domain_by_name table.

There will be additional call to Cassandra, but we have cache, and most of time, we do get, not update